### ISSUE TITLE
SIO modelling of diagnoses

### ISSUE THREAD

**Issue by markwilkinson** [LUMC-BioSemantics/ERN-common-data-elements#3]
In the "disease history and diagnosis" there are some problems with the use of SIO v.v. processes vs entities, with some (important) entities missing...

I would propose something like:

DiagProc(DP) sio:has-participant Patient(P)
DP a sio:medical-diagnosis (process!)

P sio:has-attribute D
D a sio:diagnosis (entity!)
D sio:has-measurement-value DMV
P ero:has_diagnosis DMV

DP sio:has-output DMV
DMV a [diagnosis_code]

DMV sio:has-attribute AgeAtDiagnosis(AAD)
AAD sio:has-measurement-value AADMV
AADMV a [omit/loinc/ncit code]
AADMV has-value  "27"  units [uo:years]

**Comment by markwilkinson** [comment]
a process node is important because it may need to be annotated independently of the diagnosis itself (process may be done by a machine, or an assistant, while diagnosis may be made by one specific expert, or by another machine)

**Comment by markwilkinson** [comment]
OK, I have thought about this some more...

DiagProc  a  sio:medical-diagnosis
DiagProc sio:has-participant  P
P  a  Person (with patient role)

DiagProc sio:has-output Val
P  sio:has-attribute Diag
P  ero:has diagnosis  Diag

Diag  a   sio:attribute
Diag  sio:has-measurement-value  Val

Val rdf:type <http://www.orpha.net/ORDO/Orphanet_586>
Val rdf:type sio:diagnostic-opinion
Val  sio:has-value  "Orphanet_586"
Val  rdfs:label   "Cystic Fibrosis"
<http://www.orpha.net/ORDO/Orphanet_586>   a     edam:disease-identifier

Diag  sio:has-attribute AAT
AAT  a   efo:age-at-diagnosis
Diag  sio:has-meas-value  MVAAT
MVAAT  sio:has-value   10
MVAAT  sio:has-unit   "years"
   ...or...
MVAAT   sio:has-value   <antenatal/at-birth/undetermined (by code)>

**Comment by markwilkinson** [comment]
Pffff.,... no..... I still don't like it.  Age at diagnosis isn't a property of the diagnosis, it's a property of the patient.

**Comment by rajaram5** [comment]
Should be model diagnosis as an event and attach person as a participant of the event?

**Comment by markwilkinson** [comment]
yes.  sio:medical-diagnosis is an Activity, which has participant Person.  Person has attribute, and activity generates attribute-value.

**Comment by markwilkinson** [comment]
I'm still not 100% sure that the attribute value should be ontologically typed as the Orphanet disease... I need to talk to Michel about this.

**Comment by markwilkinson** [comment]
I just fixed the model - DiagProc was pointing to Diag, but it should be pointing to Val
